pkg/controller/inferenceservice/controller.go

```diff
@@ -152,6 +152,7 @@ func (r *ReconcileService) Reconcile(request reconcile.Request) (reconcile.Resul
 
 	if err = r.updateStatus(isvc); err != nil {
 		r.Recorder.Eventf(isvc, v1.EventTypeWarning, "InternalError", err.Error())
+		return reconcile.Result{}, err
```

Comment:
when fail to updateStatus somehow, we should `return err` so that the change of this cycle will be requeue